src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php

```diff
@@ -398,7 +398,7 @@ private function registerSwaggerConfiguration(ContainerBuilder $container, array
         $container->setParameter('api_platform.enable_swagger_ui', $config['enable_swagger_ui']);
         $container->setParameter('api_platform.enable_re_doc', $config['enable_re_doc']);
         $container->setParameter('api_platform.swagger.api_keys', $config['swagger']['api_keys']);
-        $container->setParameter('api_platform.swagger_ui.extra_configuration', $config['openapi']['swagger_ui_extra_configuration'] ?? $config['swagger']['swagger_ui_extra_configuration']);
+        $container->setParameter('api_platform.swagger_ui.extra_configuration', array_merge($config['openapi']['swagger_ui_extra_configuration'], $config['swagger']['swagger_ui_extra_configuration']));
```

Comment:
```suggestion
         $container->setParameter('api_platform.swagger_ui.extra_configuration', $config['openapi']['swagger_ui_extra_configuration'] ?: $config['swagger']['swagger_ui_extra_configuration']);
```

Should do it if they're always defined. I don't like the confusion array_merge brings here.

Reply:
Ok then i suggest adding an exception
```
        if ($config['openapi']['swagger_ui_extra_configuration'] && $config['swagger']['swagger_ui_extra_configuration']) {
            throw new RuntimeException('You can not set "swagger_ui_extra_configuration" twice - in "openapi" and "swagger" section.');
        }
```
before `$container->setParameter(...);`

Reply:
Maybe we should also ask @alanpoulain what was the reason for [duplicating](https://github.com/api-platform/core/pull/3731/commits/288d0b297d6e4e94a931e87b170109bfe070addc) the `swagger_ui_extra_configuration` config in both `swagger` and `openapi`?

Reply:
It's mainly for backward compatibility reasons, since Swagger was the previous name of OpenAPI.
IMO it should be in the `openapi` section, but it makes also sense to have it in the `swagger` section since there is already the `api_keys` parameter.

Reply:
But the `api_keys` and `versions` configs are in `swagger` *only*, not duplicated in `openapi`... so doesn't that (duplication) create unnecessary confusion?

Reply:
Yes you're probably right. Should we just add it in `swagger` then?

Reply:
> But the api_keys and versions configs are in swagger only, not duplicated in openapi... so doesn't that (duplication) create unnecessary confusion?

But this should be the case imo